src/kili/services/export/format/coco/__init__.py

```diff
@@ -96,19 +96,30 @@ def _save_assets_export(
         """
         Save the assets to a file and return the link to that file
         """
-        for job_name, job in self.project_json_interface["jobs"].items():
-            if self._is_job_compatibile(job):
-                _convert_kili_semantic_to_coco(
-                    job_name=job_name,
-                    assets=assets,
-                    output_dir=Path(output_directory) / self.project_id,
-                    job=job,
-                    title=self.project_title,
-                    project_input_type=self.project_input_type,
-                    annotation_modifier=annotation_modifier,
-                )
-            else:
-                self.logger.warning(f"Job {job_name} is not compatible with the COCO format.")
+        if self.split_option == "split":
+            for job_name, job in self.project_json_interface["jobs"].items():
+                if self._is_job_compatibile(job):
+                    _convert_kili_semantic_to_coco(
+                        jobs={job_name: job},
+                        assets=assets,
+                        output_dir=Path(output_directory) / self.project_id,
+                        title=self.project_title,
+                        project_input_type=self.project_input_type,
+                        annotation_modifier=annotation_modifier,
+                        merged=False,
+                    )
+                else:
+                    self.logger.warning(f"Job {job_name} is not compatible with the COCO format.")
+        else:  # merged
+            _convert_kili_semantic_to_coco(
+                jobs=self.project_json_interface["jobs"],
+                assets=assets,
+                output_dir=Path(output_directory) / self.project_id,
+                title=self.project_title,
+                project_input_type=self.project_input_type,
+                annotation_modifier=annotation_modifier,
+                merged=True,
+            )
```

Comment:
for the merged call to _convert_kili_semantic_to_coco, we should still check for compatible jobs I think